### ISSUE TITLE
Analyze the behavior of B depending on the n of active O

### ISSUE THREAD

**Issue by leszko** [livepeer/go-livepeer#2314]
We're preparing to approach increasing the number of active Os in the round. For this purpose we need to measure if B can handle the increase number of Os. In other words, we need to measure what happens with the discovery when we increase the number of Os, for example, from `100` to `10000`.

**Comment by leszko** [comment]
## TL;DR

If we want to increase the number of active orchestrators, we need to address how to get the transcoder pool from the chain in [client.TranscoderPool()](https://github.com/livepeer/go-livepeer/blob/master/eth/client.go#L930). Additially, consider parallelizing iterating over all orchestrators in [orchestratorwatcher.handleRoundEvent()](https://github.com/livepeer/go-livepeer/blob/77450652be6d695a52effb87cfe94aad7f42c4ae/eth/watchers/orchestratorwatcher.go#L188).

Other parts of the code should work well, at least if we increase 10x the number of active orchestrators. Though, we may want to think if we really want to send so many HTTP requests to get Orchestrator Info.

## Results

Here are the parts of the code that uses the active operator pool and the impact if we increase their number by the factor of 10.

| code                                     | seq / par  | context                                               | time              | 10x time       |
| ---------------------------------------- | ---------- | ----------------------------------------------------- | ----------------- | -------------- |
| `client.TranscoderPool()`                | sequential | used during startup and before initializing the round | 113s (1.25s each) | 1125s (19 min) |
| `db_discovery.cacheOrchestratorStake()`  | parallel   | used during startup                                   | 665ms             | 1.4s           |
| `db_discovery.cacheDBOrchs()`            | parallel   | used during startup and every 1h to refresh           | 3s                | 3s             |
| `discovery.GetOrchestrators()`           | parallel   | used during each O pool session refresh               | 1s                | 1s             |
| `orchestratorwatcher.handleRoundEvent()` | sequential | used after a new round                                | 20s (0.22s each)  | 198s (3.3 min) |

Apart from the execution time, I checked the CPU and MEM. With the increase of 10x, there were some spikes in the CPU usage (probably related to the number of HTTP requests), however nothing critical.

## Comments

We definitely need to address getting the orchestrator pool from the chain, since it's currently done sequentially and already takes a lot of time. All the parallel parts does not necessarily need to be addressed, at least if we want to increase the number of orchestrators by the factor of 10x (up to 1000 active ochestrators). However, I'd still consider optimizing the discovery part, because sending so many HTTP requests may not be necessary and can cause some impact on CPU and network usage (on both B and O).

**Comment by leszko** [comment]
Closing in favor of an epic: https://github.com/livepeer/go-livepeer/issues/2324
